Updated the creating a release section of the developers manual. (#18623)
visit-dav/visit: src/doc/dev_manual/CreateRelease.rst

Reviewer: Just leaving an overall comment on quick read of first few sections. It feels like we are using "the release" as a name for a few different things here which I think can lead to some confusion.

Overview
--------

> When we put out a new release we create create two initial assets and tag the release candidate branch.

Reviewer 2: ```suggestion
When we put out a new release we create two initial assets and tag the release candidate branch.
```

1) A source code tar file that includes a ``build_visit`` script that references the tagged release.
2) A unified ``build_visit`` script that references the tagged release and includes the checksum for the source code tar file.

Creating the release
--------------------

We will describe creating a release by way of example using the steps used to create the 3.3.3 release.

> 

Reviewer: ```suggestion

Tagging the release
~~~~~~~~~~~~~~~
```

Commit a change that references the tagged release, which in our case is 3.3.3. ::

    git checkout 3.3RC
    git pull
    git checkout -b task/brugger1/2023_03_30_build_visit
    vi src/tools/dev/scripts/build_visit
    git add src/tools/dev/scripts/build_visit
    git commit -m "Temporarily set release in build_visit to v3.3.3."
    git push --set-upstream origin task/brugger1/2023_03_30_build_visit
    git branch -D task/brugger1/2023_03_30_build_visit

At this point you are ready to tag the release. ::

    git checkout 3.3RC
    git pull
    git tag v3.3.3
    git push origin v3.3.3

> 

Reviewer: ```suggestion

Creating first two release assets: source tarball and build_visit script
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reviewer: FYI...it looks like GitHub's **Suggest change** feature does not use fixed width fonts and so in cases like this, I've made an error in making heading underline match number of chars in heading.

Now you are ready to create the distribution tar file. ::

    src/tools/dev/scripts/visit-dist visit3.3.3

Now you are ready to create the unified ``build_visit`` script. ::

    src/tools/dev/scripts/build_visit --write-unified-file build_visit3_3_3

Now we revert the ``build_visit`` script on the 3.3RC branch to point at the 3.3RC. ::

    git checkout -b task/brugger1/2023_03_30_build_visit_v2
    vi src/tools/dev/scripts/build_visit
    git add src/tools/dev/scripts/build_visit
    git commit -m "Restore the release in build_visit to 3.3RC."
    git push --set-upstream origin task/brugger1/2023_03_30_build_visit_v2
    git checkout 3.3RC
    git branch -D task/brugger1/2023_03_30_build_visit_v2

> Creating the release at GitHub
> ------------------------------

Reviewer: ```suggestion
Creating the release assets page at GitHub
-------------------------------------------
```

> Now we are ready to create a new release at GitHub.

Reviewer: ```suggestion
Now we are ready to create a new release assets page at GitHub.
```

If you go to GitHub and go to the *Releases* tab you can create the new release.
Click on *Draft a new release* to bring up the form to create a new release. 

.. figure:: images/Release-GitHubStep1.png

   Drafting the new release.

Now you can choose the tag for the release.
Click on *Choose a tag* and select the ``v3.3.3`` tag.

.. figure:: images/Release-GitHubStep2.png

   Choosing the tag.

> Now you can describe the release.

Reviewer: ```suggestion
Now you can describe the release.
Occasionally, we update the description of a release after the release is completed to deal with minor changes post-release.
```

Enter ``v3.3.3`` for the title and add the description as shown below.

.. figure:: images/Release-GitHubStep3.png

   Describing the release.

At this point you can go to the bottom of the window and click on *Publish release*.

Your newly created release will now appear.

.. figure:: images/Release-GitHubStep4.png

   The newly created release.

Now you can edit the release and add the unified ``build_visit`` script and the source code tar file.

> You should only *Save* the release and not publish the release.

Reviewer: Why do we specify `not publish` here but have `click the Publish relase` on line 78 above?
```suggestion
Do not *Publish* the release as you add assets.
You should just *Save* the release.
```

If you publish the release, a notification will go out to everyone watching the VisIt_ repository and it will show up as the latest release on the releases tab.

> You should wait until you have most, if not all, the assets before publishing the release.

Reviewer 2: ```suggestion
You should wait until you have most, if not all, of the assets before publishing the release.
```

If you are releasing a new patch release you will need to add a column on the left side of the table.
The most recent release is added as the first column in the table.
In each case you will should use an existing table as an example to use for adding the new release.
Each release also has a series of shortcuts for each of the release assets.
You can copy an existing series of shortcuts and update them for the current release.
If you don't have all of the release assets added to the release you should use *Coming soon!* as a substitute for the link.
If you don't do this, people will follow the broken links and report it to you.
You should only commit the changes once the release has been published so that the links actually point to something.

Creating a blog entry for the new release
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

We create a new blog entry for each release.
The blog entries are located in ``_posts``.
Copy one of the existing blog posts for a release that is closest to the type of release you are making, either a patch release or a minor release, as a starting point.
Remove or update any version specific content from the new blog post.

> Patch releases list the number of bug fixes and ennncements along with a teaser of an intersting enhancement.

Reviewer 2: ```suggestion
Patch releases list the number of bug fixes and enhancements along with a teaser of an interesting enhancement.
```

> Minor relases also contain a teaser followed by the two or three major enhancements in the release.

Reviewer 2: ```suggestion
Minor releases also contain a teaser followed by the two or three major enhancements in the release.
```

Creating the release notes
~~~~~~~~~~~~~~~~~~~~~~~~~~

The website contains a copy of the release notes.
The release notes in the VisIt_ repository is written in raw html and the website is written in Markdown.
The release notes are located in ``pages/releases``.
Copy one of the release notes files that is closest to the type of release you are making, either a patch release or a minor release, as a starting point.
Remove or update any version specific content from the new release notes file.
Copy the raw html from the VisIt_ repository release notes into the Markdown release notes file and convert the raw html into Markdown.
Commit the changes when you are finished.

Updating the release table
~~~~~~~~~~~~~~~~~~~~~~~~~~

The website contains a page with a series of tables, one for each minor release series with links to the assets for the releases.
The tables are located in ``pages/releases-as-tables.md``.
If you are releasing a new minor release series you will need to add a new table.
If you are releasing a new patch release you will need to add a column on the left side of the table.
The most recent release is added as the first column in the table.

> In each case you will should use an existing table as an example to use for adding the new release.

Reviewer 3: ```suggestion
In each case you should use an existing table as an example for adding the new release.
```
